code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilderTest.kt

```diff
@@ -0,0 +1,239 @@
+import com.adobe.marketing.mobile.notificationbuilder.internal.PendingIntentUtils
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils.cacheImages
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils.getAssetCacheLocation
+import com.adobe.marketing.mobile.notificationbuilder.internal.PushTemplateImageUtils.getCachedImage
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.ManualCarouselNotificationBuilder.getCarouselIndices
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.ManualCarouselNotificationBuilder.populateFilmstripCarouselImages
+import com.adobe.marketing.mobile.notificationbuilder.internal.builders.ManualCarouselNotificationBuilder.populateManualCarouselImages
+import com.adobe.marketing.mobile.notificationbuilder.internal.extensions.setRemoteViewClickAction
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.ManualCarouselPushTemplate
+import com.adobe.marketing.mobile.notificationbuilder.internal.templates.provideMockedManualCarousalTemplate
+import io.mockk.Runs
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockkClass
+import io.mockk.mockkObject
+import io.mockk.verify
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+import org.robolectric.annotation.Config
+
+@RunWith(RobolectricTestRunner::class)
+@Config(sdk = [31])
+class ManualCarouselNotificationBuilderTest {
+
+    private lateinit var context: Context
+    private lateinit var pushTemplate: ManualCarouselPushTemplate
+    private lateinit var expandedLayout: RemoteViews
+    private lateinit var trackerActivityClass: Class<out Activity>
+    private lateinit var broadcastReceiverClass: Class<out BroadcastReceiver>
+
+    @Before
+    fun setUp() {
+        context = RuntimeEnvironment.getApplication()
+        pushTemplate = provideMockedManualCarousalTemplate(false)
+        trackerActivityClass = mockkClass(Activity::class, relaxed = true).javaClass
+        broadcastReceiverClass = mockkClass(BroadcastReceiver::class, relaxed = true).javaClass
+        mockkObject(PushTemplateImageUtils)
+        mockkObject(BasicNotificationBuilder)
+        mockkObject(PendingIntentUtils)
+        expandedLayout = mockkClass(RemoteViews::class)
+    }
+
+    @Test
+    fun `construct returns NotificationCompat Builder for valid inputs`() {
+        val result = ManualCarouselNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        assertNotNull(result)
```

Comment:
can we verify the returned result is of the expected class `ManualCarouselNotificationBuilder`?

Reply:
my mistake, a NotificationBuilder is returned in this case